src/svgdevicecontext.cpp

```diff
@@ -151,19 +152,15 @@ void SvgDeviceContext::Commit(bool xml_declaration)
     m_committed = true;
 }
 
-void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::string gId, bool prepend)
+void SvgDeviceContext::StartGraphic(Object *object, std::string gClass, std::string gId, bool primary, bool prepend)
 {
     std::string baseClass = object->GetClassName();
-    std::transform(baseClass.begin(), baseClass.begin() + 1, baseClass.begin(), ::tolower);
-    if (gClass.length() > 0) {
-        baseClass.append(" " + gClass);
-    }
 
     if (object->HasAttClass(ATT_TYPED)) {
         AttTyped *att = dynamic_cast<AttTyped *>(object);
         assert(att);
         if (att->HasType()) {
-            baseClass.append(" " + att->GetType());
+            gClass.append(gClass.empty() ? baseClass : baseClass + att->GetType());
```

Comment:
Why is `baseClass` given only when there a `@type`? This should be unrelated, shouldn't it?

Reply:
Absolutely right. Tested more thoroughly now.